pkg/dispatcher/dispatcher/dispatcher.go

```diff
@@ -228,3 +243,67 @@ func (d *DispatcherImpl) closeConsumerGroup(subscriber *SubscriberWrapper) {
+	d.Logger.Debug("New ConfigMap Received", zap.String("configMap.Name", configMap.ObjectMeta.Name))
+
+	// The new configmap could technically have changes to the eventing-kafka section as well as the sarama
+	// section, but none of those matter to a currently-running Dispatcher, so those are ignored here
+	// (which avoids the necessity of calling env.GetEnvironment and env.VerifyOverrides).  If those settings
+	// are needed in the future, the environment will also need to be re-parsed here.
+
+	newConfig := util.NewSaramaConfig()
+
+	// In order to compare configs "without the producer section" we use a known-base config
+	// and ensure that those sections are always the same.  The reason we can't just use, for example,
+	// sarama.Config{}.Producer as the empty struct is that the Sarama calls to create objects like ConsumerGroup
+	// still verify that certain fields (such as Producer.MaxMessageBytes) are nonzero and fail otherwise.
+	emptyProducer := newConfig.Producer
+
+	// If there aren't any consumer-specific differences between the current config and the new one,
+	// then just log that and move on; do not restart the ConsumerGroups unnecessarily.
+
+	err := commonconfig.MergeSaramaSettings(newConfig, configMap)
+	if err != nil {
+		d.Logger.Error("Unable to merge sarama settings", zap.Error(err))
+		return nil
+	}
+
+	// Don't care about Producer section; everything else is a change that needs to be implemented.
+	newConfig.Producer = emptyProducer
+
+	if d.SaramaConfig != nil {
+		// Some of the current config settings may not be overridden by the configmap (username, password, etc.)
+		util.UpdateSaramaConfig(newConfig, d.SaramaConfig.ClientID, d.SaramaConfig.Net.SASL.User, d.SaramaConfig.Net.SASL.Password)
+
+		// Create a shallow copy of the current config so that we can empty out the Producer before comparing.
+		configCopy := d.SaramaConfig
+
+		// The current config should theoretically have these sections zeroed already because Reconfigure should have been passed
+		// a newConfig with the structs empty, but this is more explicit as to what our goal is and doesn't hurt.
+		configCopy.Producer = emptyProducer
+		if commonconfig.SaramaConfigEqual(newConfig, configCopy) {
+			d.Logger.Info("No Consumer changes detected in new config; ignoring")
+			return nil
+		}
+	}
+
+	return d.reconfigure(newConfig)
+}
+
+// Reconfigure takes a new sarama.Config struct and applies the updated settings,
+// restarting the Dispatcher/ConsumerGroups if required
+func (d *DispatcherImpl) reconfigure(config *sarama.Config) Dispatcher {
+	d.Logger.Info("Configuration received; applying new Consumer settings")
+
+	// "Reconfiguring" the Dispatcher involves creating a new one, but we can re-use some
+	// of the original components, such as the list of current subscribers
+	d.Shutdown()
+	d.DispatcherConfig.SaramaConfig = config
+	newDispatcher := NewDispatcher(d.DispatcherConfig)
```

Comment:
Why do you need to set the `d.DispatcherConfig.SaramaConfig` when you're creating a new one and that one will be garbage collected ?